src/main/java/com/americanexpress/span/utility/PropertyConfiguration.java

```diff
@@ -0,0 +1,17 @@
+package com.americanexpress.span.utility;
+
+import com.americanexpress.span.constants.SPANConstants;
+
+/***
+ *  This functional interface is used to assign application profile.
+ *  User can override default method if the file name is predefined.
+ */
+
+public interface PropertyConfiguration {
+
+    default String getFileName() {
+        return SPANConstants.EMPTY_STRING;
+    }
+
+    String getAppProfile();
```

Comment:
I made it as functional interface so if application which uses this framework can use lambda expression if it want to only pass the profile. Main purpose of this was less coding for app team.